Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -201,9 +204,13 @@ def update(self):
                 self.x=self.constraint.proximal(self.x, tau=1)
 
     def update_objective(self):
-        r"""Returns the objective
+        r"""Returns the weighted least squares objective
 
-        .. math:: \frac{1}{2}\|A x - b\|^{2}
+        .. math:: \frac{1}{2}\|A x - b\|^{2}_{M}
 
         """
-        self.loss.append(0.5*self.r.squared_norm())
+        y = self.operator.direct(self.x)
+        y.subtract(self.data, out = y)
+        wy=self.M.multiply(y)
+        self.objective.append(0.5* y.dot(wy))
```

Comment:
Do we want the 0.5? Could be omitted for (slight) speed-up and objective function rewritten to omit this. Should think about consistency with defaults in related operations including LeastSquares and CGLS though.

Reply:
There isn't a 0.5 in CGLS (it uses   \min || A x - b ||^2_2) and the default for least squares is 1.0. So in some ways, SIRT would be odd to include the 0.5. Mathematically, it is optimising the same thing. Shall we remove it?